### DISCUSSION TITLE
Unable to install ImageMagick 7.1.1-15 on AlmaLinux 8

### DISCUSSION THREAD

**Discussion by TheOldMan2000** [ImageMagick/ImageMagick#6553] [Help]
### ImageMagick version

7.1.1-15

### Operating system

Linux

### Operating system, version and so on

AlmaLinux v8.8.0 STANDARD virtuozzo WHM/Cpanel 112.0.5

### Description

Hi,

I’ve just moved to an AlmaLinux 8 VPS with root access (6GB 128 vcores), but I can’t seem to install ImageMagick 7. It complains about missing dependencies. 

This may be my bad, it’s the first install since moving from CentOS 7.9.

The website states to obtain missing dependencies from EPEL, so I have enabled EPEL power tools and have been able to install Glibc 2.28-225.el8, however the dependencies seem to require earlier and later versions of glibc and other libraries.

Thank you!

### Steps to Reproduce

```
[root@vps ~]# dnf config-manager --set-enabled powertools
[root@vps ~]# dnf install epel-release

AlmaLinux 8 - PowerTools                                                       14 kB/s | 4.1 kB     00:00    
Package epel-release-8-19.el8.noarch is already installed.
Dependencies resolved.
Nothing to do.
Complete!
```

```
[root@vps ~]# wget https://imagemagick.org/archive/linux/CentOS/x86_64/ImageMagick-libs-7.1.1-15.x86_64.rpm
--2023-08-10 17:04:53--  https://imagemagick.org/archive/linux/CentOS/x86_64/ImageMagick-libs-7.1.1-15.x86_64.rpm
Resolving imagemagick.org (imagemagick.org)... 50.251.58.13
Connecting to imagemagick.org (imagemagick.org)|50.251.58.13|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2513035 (2.4M) [application/x-rpm]
Saving to: ‘ImageMagick-libs-7.1.1-15.x86_64.rpm’

ImageMagick-libs-7.1.1-15.x 100%[=========================================>]   2.40M  3.42MB/s    in 0.7s    

2023-08-10 17:04:54 (3.42 MB/s) - ‘ImageMagick-libs-7.1.1-15.x86_64.rpm’ saved [2513035/2513035]

```

```
[root@vps ~]# wget https://imagemagick.org/archive/linux/CentOS/x86_64/ImageMagick-7.1.1-15.x86_64.rpm
--2023-08-10 17:05:17--  https://imagemagick.org/archive/linux/CentOS/x86_64/ImageMagick-7.1.1-15.x86_64.rpm
Resolving imagemagick.org (imagemagick.org)... 50.251.58.13
Connecting to imagemagick.org (imagemagick.org)|50.251.58.13|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 74701 (73K) [application/x-rpm]
Saving to: ‘ImageMagick-7.1.1-15.x86_64.rpm’

ImageMagick-7.1.1-15.x86_64 100%[=========================================>]  72.95K  --.-KB/s    in 0.04s   

2023-08-10 17:05:18 (1.61 MB/s) - ‘ImageMagick-7.1.1-15.x86_64.rpm’ saved [74701/74701]

```

```
[root@vps ~]# rpm -Uvh ImageMagick-libs-7.1.1-15.x86_64.rpm
error: Failed dependencies:
        libc.so.6(GLIBC_2.33)(64bit) is needed by ImageMagick-libs-7.1.1-15.x86_64
        libc.so.6(GLIBC_2.34)(64bit) is needed by ImageMagick-libs-7.1.1-15.x86_64
        libgs.so.10()(64bit) is needed by ImageMagick-libs-7.1.1-15.x86_64
        libjxl.so.0.7()(64bit) is needed by ImageMagick-libs-7.1.1-15.x86_64
        libjxl.so.0.7(JXL_0)(64bit) is needed by ImageMagick-libs-7.1.1-15.x86_64
        libjxl_threads.so.0.7()(64bit) is needed by ImageMagick-libs-7.1.1-15.x86_64
        libjxl_threads.so.0.7(JXL_0)(64bit) is needed by ImageMagick-libs-7.1.1-15.x86_64
        libm.so.6(GLIBC_2.29)(64bit) is needed by ImageMagick-libs-7.1.1-15.x86_64
        libm.so.6(GLIBC_2.35)(64bit) is needed by ImageMagick-libs-7.1.1-15.x86_64
        libraw_r.so.23()(64bit) is needed by ImageMagick-libs-7.1.1-15.x86_64
        ImageMagick-libs(x86-64) = 0:6.9.12.86-1.el8 is needed by (installed) ImageMagick-0:6.9.12.86-1.el8.x86_64
        libMagickCore-6.Q16.so.7()(64bit) is needed by (installed) ImageMagick-0:6.9.12.86-1.el8.x86_64
        libMagickWand-6.Q16.so.7()(64bit) is needed by (installed) ImageMagick-0:6.9.12.86-1.el8.x86_64

```

```
[root@vps ~]# rpm -Uvh ImageMagick-7.1.1-15.x86_64.rpm
error: Failed dependencies:
        ImageMagick-libs(x86-64) = 7.1.1-15 is needed by ImageMagick-7.1.1-15.x86_64
        libMagickCore-7.Q16HDRI.so.10()(64bit) is needed by ImageMagick-7.1.1-15.x86_64
        libMagickWand-7.Q16HDRI.so.10()(64bit) is needed by ImageMagick-7.1.1-15.x86_64
        libMagickWand-7.Q16HDRI.so.10(VERS_10.0)(64bit) is needed by ImageMagick-7.1.1-15.x86_64
        libc.so.6(GLIBC_2.34)(64bit) is needed by ImageMagick-7.1.1-15.x86_64

```

**Comment by jhetoro** [comment]
hello, could you solve it?

**Reply by TheOldMan2000** [reply]
No not yet @jhetoro still trying!

```
dnf config-manager --set-enabled epel
[root@vps~]# dnf update -y
Extra Packages for Enterprise Linux 8 - x86_64                                356 kB/s |  27 kB     00:00    
Dependencies resolved.
Nothing to do.
Complete!
[root@vps ~]# dnf provides ImageMagick
Last metadata expiration check: 0:00:14 ago on Sat 19 Aug 2023 01:48:25 PM BST.
**ImageMagick-6.9.12.86-1**.el8.x86_64 : An X application for displaying and manipulating images
Repo        : epel
Matched from:
License     : ImageMagick
```

**Comment by TheOldMan2000** [comment]
Hi @jhetoro  Finally got this working:

Almalinux 8 

````

wget https://download.imagemagick.org/archive/releases/ImageMagick-7.1.1-15.tar.xz

mv ImageMagick-7.1.1-15 ImageMagick-7

cd ImageMagick-7

less Install-unix.txt <- optional this was helpful

./configure

sudo make install

make check

Eventually shows following output:

Testsuite summary for ImageMagick 7.1.1-15
============================================================================
# TOTAL: 87
# PASS:  87
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[3]: Leaving directory '/root/ImageMagick-7'
make[2]: Nothing to be done for 'check-local'.
make[2]: Leaving directory '/root/ImageMagick-7'
make[1]: Leaving directory '/root/ImageMagick-7'

Quick list of supported formats:

identify -list format

````

Regards
G.

**Reply by fmw42** [reply]
Are you sure?  What do you get from 

`magick -version`

Does it show all the needed delegates?  You have to install delegates before installing IM

**Reply by mehmetizmirlioglu** [reply]
Thanks. It is worked!

**Comment by TheOldMan2000** [comment]
Hi @fmw42 

Well I think it’s okay, I installed the PHP Imagick extension afterwards and it shows up in phpinfo.

I ran your test and got:

```
magick -version
Version: ImageMagick 7.1.1-15 Q16-HDRI x86_64 21298 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib fontconfig freetype jng jpeg lzma png tiff webp x xml zlib
Compiler: gcc (8.5)
```

**Reply by fmw42** [reply]
yes looks fine

**Reply by TheOldMan2000** [reply]
Thanks, now I’m wondering why it’s working, though! I tried and failed to install the latest glibc package it previously stated it needed when I tried installing from the repository.

**Comment by manos2000** [comment]
Hi there!
Tried today to install ImageMagick 7.1.1-31 at Almalinux 8.9 (with plesk)
Used phpize from php 8.2 but at whatever version same message that packet is missing config.m4, you did not have this problem at ./configure?
With rpm files had exactly same issues like you with missing libraries, so finally i enabled remi repo and installed it from there

dnf provides ImageMagick returns correctly ImageMagick 7.1.1-31

Then i compiled, starting with phpize of php 8.2, imagick, and at php -i | grep ImageMagick have correct output imagick 3.7 compiled and using ImageMagick 7.1.1-31 o everything works fine, i copied the imagick.so file of modules directory to plesk php 8.2 modulesdirectory with another name and changed accordingly imagick.ini file.

How did you bypass the missing config.m4 at ./configure at ImageMagick source folder?

Thanks!

**Comment by ProPixelMedia** [comment]
I'm having similar problems.

Been at this for about 2 hours now, read about 20 articles on Google. Fired everything I had into the terminal and its still not working. Restarted Apache and PHP-FPM several times.

My Server configuration: CentOS 7 PHP Version 7.4.33 VPS hosting

Imagemagick 6.9 is currently working but I need to install version 7 to access the newer functions. Got the newer 7.1.1 version "installed", but PHP is still using the old version. When I run the command:

convert --version

Version: ImageMagick 7.1.1-33 Q16-HDRI x86_64 22263 https://imagemagick.org/ Copyright: (C) 1999 ImageMagick Studio LLC License: https://imagemagick.org/script/license.php Features: Cipher DPC HDRI OpenMP(3.1) Delegates (built-in): bzlib fontconfig freetype jng jpeg lzma png tiff x xml zlib Compiler: gcc (4.8)

output from phpinfo(): imagick module version 3.7.0 Imagick compiled with ImageMagick version ImageMagick 6.9.10-68 Q16 x86_64 2024-01-12

Any ideas how I can force PHP to use the newer version? Tried reading a bunch of articles about how to delete Version 6.9, but some of the commands aren't working. I believe alot of the information out there is for Ubuntu while I'm running CentOS and that's why basic commands aren't working, such as: sudo phpdismod imagick

sudo apt remove php-imagick

When i run those commands, I get errors like this: sudo: apt: command not found

Been at this for about 2 hours now, read about 20 articles on Google. Fired everything I had into the terminal and its still not working. Restarted Apache and PHP-FPM several times.

[EDIT] I've upgraded my server to AlmaLInux 9 (with cPanel), and mostly the same problem. I'm running PHP 8.1

I'm still stuck with Version 6.9. Trying to get PHP to use version 7.1

Here's what comes up with I do print_r(imagick::getVersion()); in PHP: [versionNumber] => 1693 [versionString] => ImageMagick 6.9.13-11 Q16 x86_64 18407 https://legacy.imagemagick.org/

Here's what comes up with I do convert --version in the shell: WARNING: The convert command is deprecated in IMv7, use "magick"

Version: ImageMagick 7.1.1-33 Q16-HDRI x86_64 22263 https://imagemagick.org/ Copyright: (C) 1999 ImageMagick Studio LLC License: https://imagemagick.org/script/license.php Features: Cipher DPC HDRI OpenMP(4.5) Delegates (built-in): bzlib fontconfig freetype jng jpeg lzma png tiff webp x xml zlib zstd Compiler: gcc (11.4)

Here's what comes up with I do magick -version in the shell: Version: ImageMagick 7.1.1-33 Q16-HDRI x86_64 22263 https://imagemagick.org/ Copyright: (C) 1999 ImageMagick Studio LLC License: https://imagemagick.org/script/license.php Features: Cipher DPC HDRI OpenMP(4.5) Delegates (built-in): bzlib fontconfig freetype jng jpeg lzma png tiff webp x xml zlib zstd Compiler: gcc (11.4)

So it seems like its installed on the server, but PHP still using the wrong version.

Spent another 3 hours on it last night installing and re-installing everything I could find. Is it true that I also need to install phpize and php-devel to get this to work? both of those software packages have been difficult for me to install. The server keeps rejecting my attempts to get them installed. I am at the ponit where I may have to download them manually to get them to install, but i can't even find a version that is compatible with AlmaLInux 9

Any ideas?
